homeassistant/components/lookin/media_player.py

```diff
@@ -0,0 +1,215 @@
+        super().__init__(coordinator, uuid, device, lookin_data)
+        for function_name, feature in _FUNCTION_NAME_TO_FEATURE.items():
+            if function_name in self._function_names:
+                self._attr_supported_features |= feature
+        self._async_update_from_data()
+
+    @property
+    def _remote(self) -> Remote:
+        return cast(Remote, self.coordinator.data)
+
+    @property
+    def state(self) -> str | None:
+        """State of the player."""
+        return self._state
+
+    async def async_volume_up(self) -> None:
+        """Turn volume up for media player."""
+        await self._async_send_command("volup")
+
+    async def async_volume_down(self) -> None:
+        """Turn volume down for media player."""
+        await self._async_send_command("voldown")
+
+    async def async_media_previous_track(self) -> None:
+        """Send previous track command."""
+        await self._async_send_command("chdown")
+
+    async def async_media_next_track(self) -> None:
+        """Send next track command."""
+        await self._async_send_command("chup")
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        await self._async_send_command("mute")
+        self._is_muted = not self._is_muted
+
+    async def async_turn_off(self) -> None:
+        """Turn the media player off."""
+        await self._async_send_command(self._power_off_command)
+        self._state = STATE_STANDBY
+
+    async def async_turn_on(self) -> None:
+        """Turn the media player on."""
+        await self._async_send_command(self._power_on_command)
+        self._state = STATE_ON
+
+    @property
+    def name(self) -> str | None:
+        return self._remote.name
+
+    @property
+    def is_volume_muted(self) -> bool | None:
+        """Boolean if volume is currently muted."""
+        return self._is_muted
+
+    def _update_from_status(self, status: str) -> None:
+        """
+        00F0
+        0 - 0/1 on/off
+        0 - sourse
+        F - volume, 0 - muted, 1 - volume up, F - volume down
+        0 - not used
+        """
+        if len(status) != 4:
+            return
+        state = status[0]
+        mute = status[2]
+
+        self._state = state == "1" and STATE_STANDBY or STATE_ON
+        self._is_muted = mute == "0"
+
+    def _async_push_update(self, event: UDPEvent) -> None:
+        """Process an update pushed via UDP."""
+        LOGGER.debug("Processing push message for %s: %s", self.entity_id, event)
+        self._update_from_status(event.value)
+        self.coordinator.async_set_updated_data(self._remote)
+        self.async_write_ha_state()
+
+    async def _async_push_update_device(self, event: UDPEvent) -> None:
+        """Process an update pushed via UDP."""
+        LOGGER.debug("Processing push message for %s: %s", self.entity_id, event)
+        await self.coordinator.async_refresh()
+        self.async_write_ha_state()
```

Comment:
```suggestion
```

The coordinator will poll the endpoint and call `async_write_ha_state` after the data is returned so we don't need to write it ourselves

Reply:
deleted